src/hooks/useServerUpdateRequiredMonitor.ts

```diff
@@ -0,0 +1,66 @@
+import { useEntityType } from 'context/EntityContext';
+import { DraftSpecQuery } from 'hooks/useDraftSpecs';
+import { isEqual } from 'lodash';
+import { useEffect, useMemo } from 'react';
+import {
+    useResourceConfig_resourceConfig,
+    useResourceConfig_setServerUpdateRequired,
+} from 'stores/ResourceConfig/hooks';
+import { hasLength } from 'utils/misc-utils';
+import {
+    getCollectionName,
+    getCollectionNameProp,
+    getDisableProps,
+} from 'utils/workflow-utils';
+
+const useServerUpdateRequiredMonitor = (draftSpecs: DraftSpecQuery[]) => {
```

Comment:
Moved into a shared hook as a couple places needed this and needed to make sure that this was always getting set the same way.